File: app/src/main/java/com/mobile/responses/MicroServiceRestrictionsResponse.java
package com.mobile.responses;

import com.mobile.model.Alert;
import com.mobile.model.LogoutInfo;
import com.mobile.model.PopInfo;

public class MicroServiceRestrictionsResponse {


    public MicroServiceRestrictionsResponse() {
    }


    int countdown;
    String subscriptionStatus;
    boolean facebook;
    boolean has3d;
    boolean hasAllFormats;
    boolean proofOfPurchaseRequired;
    PopInfo popInfo;
    boolean hasActiveCard;
    Alert alert;
    LogoutInfo logoutInfo;

    public LogoutInfo getLogoutInfo() {
        return logoutInfo;
    }

    public boolean isSubscriptionActivationRequired() {
        return subscriptionActivationRequired;
    }

    boolean subscriptionActivationRequired;


    public int getCountdown() {
        return countdown;
    }

    public String getSubscriptionStatus() {
        return subscriptionStatus;
    }

    public boolean getFacebookPresent() {
        return facebook;
    }

    public boolean get3dEnabled() {
        return has3d;
    }

    public boolean getAllFormatsEnabled() {
        return hasAllFormats;
    }

    public boolean getProofOfPurchaseRequired() {
        return proofOfPurchaseRequired;
    }

    public PopInfo getPopInfo() {
        return popInfo;
    }

    public boolean getHasActiveCard() {
        return hasActiveCard;
    }

    public Alert getAlert() {
        return alert;
    }
}

File: app/src/main/java/com/mobile/responses/MicroServiceRestrictionsResponse.kt
package com.mobile.responses

import com.mobile.model.Alert
import com.mobile.model.LogoutInfo
import com.mobile.model.PopInfo
import com.mobile.model.canReactivate


class MicroServiceRestrictionsResponse(
        var countDown: Int = 0,
        var subscriptionStatus: SubscriptionStatus = SubscriptionStatus.UNKNOWN,
        var facebook: Boolean = false,
        var has3d: Boolean = false,
        var hasAllFormats: Boolean = false,
        var proofOfPurchaseRequired: Boolean = false,
        var popInfo: PopInfo? = null,
        var hasActiveCard: Boolean = false,
        var alert: Alert? = null,
        var logoutInfo: LogoutInfo? = null,
        var subscriptionActivationRequired:Boolean = false,
        var canReactivate: canReactivate? = null,
        var userSegments:List<Int> = emptyList()
)
